### PR TITLE
Use conditional access

### LINKED ISSUE #5591: Use conditional access
There are several parts in the code where code can be simplified with conditional access.

Found using Roslynator
https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS1146.md

### PR DESCRIPTION
**Pull request by elachlan**
Fixes #5591

### CODE REVIEW

File: src/Build/BackEnd/Components/RequestBuilder/TargetEntry.cs [hunk 1]
```diff
@@ -531,7 +531,7 @@ internal async Task ExecuteTarget(ITaskBuilder taskBuilder, BuildRequestEntry re
                                 entryForInference = null;
                                 entryForExecution.LeaveScope();
                                 entryForExecution = null;
-                                targetSuccess = (bucketResult != null) && (bucketResult.ResultCode == WorkUnitResultCode.Success);
+                                targetSuccess = (bucketResult?.ResultCode == WorkUnitResultCode.Success);
```
Comment (Forgind) [inline src/Build/BackEnd/Components/RequestBuilder/TargetEntry.cs:534]: nit:
Take away parentheses

File: src/Build/BackEnd/Components/RequestBuilder/TargetEntry.cs [hunk 2]
```diff
@@ -653,11 +647,11 @@ internal async Task ExecuteTarget(ITaskBuilder taskBuilder, BuildRequestEntry re
                 }
                 finally
                 {
-                    if (targetLoggingContext != null)
-                    {
+
```
Comment (Forgind) [inline src/Build/BackEnd/Components/RequestBuilder/TargetEntry.cs:650]: nit: Any reason for these extra spaces?

File: src/Build/BackEnd/Components/RequestBuilder/TargetUpToDateChecker.cs [hunk 3]
```diff
@@ -235,7 +235,7 @@ internal DependencyAnalysisResult PerformDependencyAnalysis
                 // 1) they do not reference any item vector
                 // 2) they reference item vectors that are not referenced by any input item
                 if ((discreteItemsInTargetOutputs.Count > 0) ||
-                    ((itemVectorsReferencedOnlyInTargetOutputs != null) && (itemVectorsReferencedOnlyInTargetOutputs.Count > 0)))
+                    ((itemVectorsReferencedOnlyInTargetOutputs?.Count > 0)))
```
Comment (Forgind) [inline src/Build/BackEnd/Components/RequestBuilder/TargetUpToDateChecker.cs:238]: nit:
Remove extra parentheses

File: src/Build/Construction/Solution/SolutionFile.cs [hunk 4]
```diff
@@ -757,7 +757,7 @@ private void ParseProject(string firstLine)
                     // We have a ProjectDependencies section.  Each subsequent line should identify
                     // a dependency.
                     line = ReadLine();
-                    while ((line != null) && (!line.StartsWith("EndProjectSection", StringComparison.Ordinal)))
+                    while ((line?.StartsWith("EndProjectSection", StringComparison.Ordinal) == false))
```
Comment (Forgind) [inline src/Build/Construction/Solution/SolutionFile.cs:760]: nit:
extra parentheses

File: src/Build/Construction/Solution/SolutionFile.cs [hunk 5]
```diff
@@ -777,7 +777,7 @@ private void ParseProject(string firstLine)
                     // projects, and contains properties that we'll need in order to call the 
                     // AspNetCompiler task.
                     line = ReadLine();
-                    while ((line != null) && (!line.StartsWith("EndProjectSection", StringComparison.Ordinal)))
+                    while ((line?.StartsWith("EndProjectSection", StringComparison.Ordinal) == false))
```
Comment (Forgind) [inline src/Build/Construction/Solution/SolutionFile.cs:780]: nit:
extra parentheses

File: src/Build/Definition/ProjectMetadata.cs [hunk 6]
```diff
@@ -132,7 +132,7 @@ public string UnevaluatedValue
             {
                 ErrorUtilities.VerifyThrowArgumentNull(value, "value");
                 Project.VerifyThrowInvalidOperationNotImported(_xml.ContainingProject);
-                ErrorUtilities.VerifyThrowInvalidOperation(_xml.Parent != null && _xml.Parent.Parent != null && _xml.Parent.Parent.Parent != null, "OM_ObjectIsNoLongerActive");
+                ErrorUtilities.VerifyThrowInvalidOperation(_xml.Parent != null && _xml.Parent.Parent?.Parent != null, "OM_ObjectIsNoLongerActive");
```
Comment (Forgind) [inline src/Build/Definition/ProjectMetadata.cs:135]: nit:
```suggestion
                ErrorUtilities.VerifyThrowInvalidOperation(_xml.Parent?.Parent?.Parent != null, "OM_ObjectIsNoLongerActive");
```

😄

File: src/Deprecated/Engine/Engine/BuildTask.cs [hunk 7]
```diff
@@ -517,11 +517,11 @@ private void MarkTaskAsDirty
             (
             )
         {
-            if (this.ParentTarget != null)
-            {
+
```
Comment (Forgind) [inline src/Deprecated/Engine/Engine/BuildTask.cs:520]: nit: extra spaces

File: src/Deprecated/Engine/Engine/Router.cs [hunk 8]
```diff
@@ -131,12 +131,12 @@ internal void PostDoneNotice(BuildRequest buildRequest)
         /// </summary>
         internal void PostDoneNotice(int nodeId, BuildResult buildResult)
         {
-            if (scheduler != null)
-            {
+               
+
```
Comment (Forgind) [inline src/Deprecated/Engine/Engine/Router.cs:135]: nit: spaces

File: src/Deprecated/Engine/Engine/Router.cs [hunk 9]
```diff
@@ -179,12 +179,12 @@ internal void PostBuildRequest(BuildRequest currentRequest, int nodeIndex)
                                     (nodeIndex, currentRequest.HandleId, currentRequest.NodeIndex,
                                      currentRequest.RequestId, cacheScope, currentRequest, null);
 
-                    if (scheduler != null)
-                    {
+
```
Comment (Forgind) [inline src/Deprecated/Engine/Engine/Router.cs:182]: nit:
spaces

File: src/Deprecated/Engine/Engine/Target.cs [hunk 10]
```diff
@@ -732,11 +732,11 @@ internal void MarkTargetAsDirty
             (
             )
         {
-            if (this.ParentProject != null)
-            {
+               
+
```
Comment (Forgind) [inline src/Deprecated/Engine/Engine/Target.cs:736]: nit: spaces

File: src/Deprecated/Engine/Engine/TargetDependencyAnalyzer.cs [hunk 11]
```diff
@@ -238,7 +238,7 @@ internal DependencyAnalysisResult PerformDependencyAnalysis
                 // 1) they do not reference any item vector
                 // 2) they reference item vectors that are not referenced by any input item
                 if ((discreteItemsInTargetOutputs.Count > 0) ||
-                    ((itemVectorsReferencedOnlyInTargetOutputs != null) && (itemVectorsReferencedOnlyInTargetOutputs.Count > 0)))
+                    ((itemVectorsReferencedOnlyInTargetOutputs?.Count > 0)))
```
Comment (Forgind) [inline src/Deprecated/Engine/Engine/TargetDependencyAnalyzer.cs:241]: nit: extra parens

File: src/Deprecated/Engine/Engine/TaskEngine.cs [hunk 12]
```diff
@@ -1018,7 +1012,7 @@ private void GatherArrayStringAndValueOutputs(Lookup lookup, TaskOutput taskOutp
                 else
                 {
                     Debug.Assert(taskOutputSpecification.IsProperty);
-                    ErrorUtilities.VerifyThrow((propertyName != null) && (propertyName.Length > 0), "Need property name.");
+                    ErrorUtilities.VerifyThrow((propertyName?.Length > 0), "Need property name.");
```
Comment (Forgind) [inline src/Deprecated/Engine/Engine/TaskEngine.cs:1015]: extra parens * most of this file

File: src/Deprecated/Engine/Engine/Target.cs [hunk 13]
```diff
@@ -226,7 +226,7 @@ bool        importedFromAnotherProject
             }
 
             // It's considered an error if a target does not have a name.
-            ProjectErrorUtilities.VerifyThrowInvalidProject((targetName != null) && (targetName.Length > 0),
```
Comment (Forgind) [inline src/Deprecated/Engine/Engine/Target.cs:229]: nit:
extra parens

File: src/Deprecated/Engine/Engine/TargetDependencyAnalyzer.cs [hunk 14]
```diff
@@ -1140,7 +1140,7 @@ private void RecordComparisonResults(string input, string output, string inputIt
         /// </returns>
         private int CompareLastWriteTimes(string path1, string path2, out bool path1DoesNotExist, out bool path2DoesNotExist)
         {
-            ErrorUtilities.VerifyThrow((path1 != null) && (path1.Length > 0) && (path2 != null) && (path2.Length > 0),
+            ErrorUtilities.VerifyThrow((path1 != null) && (path1.Length > 0) && (path2?.Length > 0),
```
Comment (Forgind) [inline src/Deprecated/Engine/Engine/TargetDependencyAnalyzer.cs:1143]: nit:
```suggestion
            ErrorUtilities.VerifyThrow(path1?.Length > 0 && path2?.Length > 0,
```

File: src/Deprecated/Engine/Errors/InternalLoggerException.cs [hunk 15]
```diff
@@ -93,10 +93,10 @@ bool initializationException
          )
             : base(message, innerException)
         {
-            ErrorUtilities.VerifyThrow((message != null) && (message.Length > 0), "Need error message.");
+            ErrorUtilities.VerifyThrow((message?.Length > 0), "Need error message.");
             ErrorUtilities.VerifyThrow(innerException != null || initializationException == true, "Need the logger exception.");
-            ErrorUtilities.VerifyThrow((errorCode != null) && (errorCode.Length > 0), "Must specify the error message code.");
```
Comment (Forgind) [inline src/Deprecated/Engine/Errors/InternalLoggerException.cs:98]: extra parens

File: src/Deprecated/Engine/Errors/RemoteErrorException.cs [hunk 16]
```diff
@@ -19,7 +19,7 @@ public sealed class RemoteErrorException : Exception
         internal RemoteErrorException(string message, Exception innerException, BuildEventContext buildEventContext)
             : base(message, innerException)
         {
-            ErrorUtilities.VerifyThrow((message != null) && (message.Length > 0), "Need error message.");
+            ErrorUtilities.VerifyThrow((message?.Length > 0), "Need error message.");
```
Comment (Forgind) [inline src/Deprecated/Engine/Errors/RemoteErrorException.cs:22]: extra parens

File: src/Deprecated/Engine/Items/BuildItem.cs [hunk 17]
```diff
@@ -647,7 +647,7 @@ public string Include
                 {
                     // If this is an evaluated item that originated from the project file, and the original
                     // item is declared using a wildcard that still matches the new item spec ...
-                    if ((ParentPersistedItem != null) && (ParentPersistedItem.NewItemSpecMatchesExistingWildcard(value)))
+                    if ((ParentPersistedItem?.NewItemSpecMatchesExistingWildcard(value) == true))
```
Comment (Forgind) [inline src/Deprecated/Engine/Items/BuildItem.cs:650]: extra parens

File: src/Deprecated/Engine/Items/BuildItemGroup.cs [hunk 18]
```diff
@@ -747,7 +744,7 @@ private void MustBeInitialized()
         /// </summary>
         private void MustHaveThisParentElement(BuildItem item)
         {
-            ErrorUtilities.VerifyThrowInvalidOperation(item != null && item.ItemElement != null && item.ItemElement.ParentNode == xml.Element, "ItemDoesNotBelongToItemGroup");
+            ErrorUtilities.VerifyThrowInvalidOperation(item?.ItemElement != null && item.ItemElement.ParentNode == xml.Element, "ItemDoesNotBelongToItemGroup");
```
Comment (Forgind) [inline src/Deprecated/Engine/Items/BuildItemGroup.cs:747]:
```suggestion
            ErrorUtilities.VerifyThrowInvalidOperation(item?.ItemElement?.ParentNode == xml.Element, "ItemDoesNotBelongToItemGroup");
```

File: src/Deprecated/Engine/Properties/BuildProperty.cs [hunk 19]
```diff
@@ -386,7 +386,7 @@ public string Name
                 else
                 {
                     // If we are not persisted, propertyName and propertyValue must not be null.
-                    ErrorUtilities.VerifyThrow((this.propertyName != null) && (this.propertyName.Length > 0) && (this.propertyValue != null),
+                    ErrorUtilities.VerifyThrow((this.propertyName?.Length > 0) && (this.propertyValue != null),
```
Comment (Forgind) [inline src/Deprecated/Engine/Properties/BuildProperty.cs:389]: nit:
extra parens

File: src/Deprecated/Engine/Shared/AssemblyLoadInfo.cs [hunk 20]
```diff
@@ -22,7 +22,7 @@ internal sealed class AssemblyLoadInfo
         /// <param name="assemblyFile"></param>
         public AssemblyLoadInfo(string assemblyName, string assemblyFile)
         {
-            ErrorUtilities.VerifyThrow(((assemblyName != null) && (assemblyName.Length > 0)) || ((assemblyFile != null) && (assemblyFile.Length > 0)),
+            ErrorUtilities.VerifyThrow(((assemblyName?.Length > 0)) || ((assemblyFile?.Length > 0)),
```
Comment (Forgind) [inline src/Deprecated/Engine/Shared/AssemblyLoadInfo.cs:25]: extra parens

File: src/Deprecated/Engine/Shared/ConfigurationInSolution.cs [hunk 21]
```diff
@@ -31,7 +31,7 @@ internal ConfigurationInSolution(string configurationName, string platformName)
             this.platformName = platformName;
 
             // Some configurations don't have the platform part
-            if ((platformName != null) && (platformName.Length > 0))
+            if ((platformName?.Length > 0))
```
Comment (Forgind) [inline src/Deprecated/Engine/Shared/ConfigurationInSolution.cs:34]: extra parens

File: src/Deprecated/Engine/Shared/FileMatcher.cs [hunk 22]
```diff
@@ -1249,8 +1249,7 @@ DirectoryExists directoryExists
             // 2) if the extension is three characters, and the filename contains the * wildcard, it matches files with longer
             //    extensions that start with the same three characters e.g. "*.htm" would match both "file.htm" and "file.html"
             bool needToEnforceExtensionLength =
-                    (extensionPart != null) &&
-                    (extensionPart.IndexOf('*') == -1)
+                    (extensionPart?.IndexOf('*') == -1)
```
Comment (Forgind) [inline src/Deprecated/Engine/Shared/FileMatcher.cs:1252]: extra parens

File: src/Deprecated/Engine/Shared/ResourceUtilities.cs [hunk 23]
```diff
@@ -121,7 +121,7 @@ internal static string FormatString(string unformatted, params object[] args)
             string formatted = unformatted;
 
             // NOTE: String.Format() does not allow a null arguments array
-            if ((args != null) && (args.Length > 0))
+            if ((args?.Length > 0))
```
Comment (Forgind) [inline src/Deprecated/Engine/Shared/ResourceUtilities.cs:124]: extra parens

File: src/Deprecated/Engine/Shared/SolutionParser.cs [hunk 24]
```diff
@@ -555,7 +549,7 @@ private void ParseProject(string firstLine)
                     // We have a ProjectDependencies section.  Each subsequent line should identify
                     // a dependency.
                     line = ReadLine();
-                    while ((line != null) && (!line.StartsWith("EndProjectSection", StringComparison.Ordinal)))
+                    while ((line?.StartsWith("EndProjectSection", StringComparison.Ordinal) == false))
```
Comment (Forgind) [inline src/Deprecated/Engine/Shared/SolutionParser.cs:552]: extra parens

File: src/Deprecated/Engine/Shared/SolutionParser.cs [hunk 25]
```diff
@@ -575,7 +569,7 @@ private void ParseProject(string firstLine)
                     // projects, and contains properties that we'll need in order to call the 
                     // AspNetCompiler task.
                     line = ReadLine();
-                    while ((line != null) && (!line.StartsWith("EndProjectSection", StringComparison.Ordinal)))
+                    while ((line?.StartsWith("EndProjectSection", StringComparison.Ordinal) == false))
```
Comment (Forgind) [inline src/Deprecated/Engine/Shared/SolutionParser.cs:572]: extra parens

File: src/Deprecated/Engine/Shared/SolutionParser.cs [hunk 26]
```diff
@@ -984,7 +978,7 @@ string property
             )
         {
             // If the incoming string starts and ends with a double-quote, strip the double-quotes.
-            if ((property != null) && (property.Length > 0) && (property[0] == '"') && (property[property.Length - 1] == '"'))
+            if ((property?.Length > 0) && (property[0] == '"') && (property[property.Length - 1] == '"'))
```
Comment (Forgind) [inline src/Deprecated/Engine/Shared/SolutionParser.cs:981]: extra parens

File: src/Deprecated/Engine/Shared/XmlUtilities.cs [hunk 27]
```diff
@@ -44,11 +44,11 @@ internal static XmlElement RenameXmlElement(XmlElement oldElement, string newEle
                 newElement.AppendChild(newChildNode);
             }
 
-            if (oldElement.ParentNode != null)
-            {
+
```
Comment (Forgind) [inline src/Deprecated/Engine/Shared/XmlUtilities.cs:47]: extra spaces

File: src/Deprecated/Engine/Shared/XmlUtilities.cs [hunk 28]
```diff
@@ -69,7 +69,7 @@ internal static string GetXmlNodeFile(XmlNode node, string defaultFile)
             string file = defaultFile;
 
             // NOTE: the XML node may not have a filename if it's purely an in-memory node
-            if ((node.OwnerDocument.BaseURI != null) && (node.OwnerDocument.BaseURI.Length > 0))
+            if ((node.OwnerDocument.BaseURI?.Length > 0))
```
Comment (Forgind) [inline src/Deprecated/Engine/Shared/XmlUtilities.cs:72]: extra parens

File: src/Deprecated/Engine/Solution/VCWrapperProject.cs [hunk 29]
```diff
@@ -99,7 +99,7 @@ string fullConfigurationName
 
             newTask.SetParameterValue("SolutionFile", solutionPath);
 
-            if ((vcbuildTargetName != null) && (vcbuildTargetName.Length > 0))
+            if ((vcbuildTargetName?.Length > 0))
```
Comment (Forgind) [inline src/Deprecated/Engine/Solution/VCWrapperProject.cs:102]: extra parens

File: src/Framework/LazyFormattedBuildEventArgs.cs [hunk 30]
```diff
@@ -196,7 +196,7 @@ private static string FormatString(CultureInfo culture, string unformatted, para
             string formatted = unformatted;
 
             // NOTE: String.Format() does not allow a null arguments array
-            if ((args != null) && (args.Length > 0))
+            if ((args?.Length > 0))
```
Comment (Forgind) [inline src/Framework/LazyFormattedBuildEventArgs.cs:199]: extra parens

File: src/MSBuild/XMake.cs [hunk 31]
```diff
@@ -2949,8 +2946,8 @@ private static string[] RemoveFilesWithExtensionsToIgnore
                                 )
         {
             // If we got to this method we should have to possible projects or solutions and some extensions to ignore
-            ErrorUtilities.VerifyThrow(((potentialProjectOrSolutionFiles != null) && (potentialProjectOrSolutionFiles.Length > 0)), "There should be some potential project or solution files");
-            ErrorUtilities.VerifyThrow(((extensionsToIgnoreDictionary != null) && (extensionsToIgnoreDictionary.Count > 0)), "There should be some extensions to Ignore");
+            ErrorUtilities.VerifyThrow(((potentialProjectOrSolutionFiles?.Length > 0)), "There should be some potential project or solution files");
```
Comment (Forgind) [inline src/MSBuild/XMake.cs:2949]: extra parens

File: src/MSBuild/XMake.cs [hunk 32]
```diff
@@ -3206,7 +3203,7 @@ ArrayList loggers
                 ConsoleLogger logger = new ConsoleLogger(verbosity);
                 string consoleParameters = "SHOWPROJECTFILE=TRUE;";
 
-                if ((consoleLoggerParameters != null) && (consoleLoggerParameters.Length > 0))
+                if ((consoleLoggerParameters?.Length > 0))
```
Comment (Forgind) [inline src/MSBuild/XMake.cs:3206]: extra parens

File: src/MSBuild/XMake.cs [hunk 33]
```diff
@@ -3273,7 +3270,7 @@ int cpuCount
             if (distributedFileLogger)
             {
                 string fileParameters = string.Empty;
-                if ((fileLoggerParameters != null) && (fileLoggerParameters.Length > 0))
+                if ((fileLoggerParameters?.Length > 0))
```
Comment (Forgind) [inline src/MSBuild/XMake.cs:3273]: extra parens

File: src/Shared/AssemblyLoadInfo.cs [hunk 34]
```diff
@@ -23,7 +23,7 @@ internal abstract class AssemblyLoadInfo : ITranslatable, IEquatable<AssemblyLoa
         /// </summary>
         internal static AssemblyLoadInfo Create(string assemblyName, string assemblyFile)
         {
-            ErrorUtilities.VerifyThrow(((assemblyName != null) && (assemblyName.Length > 0)) || ((assemblyFile != null) && (assemblyFile.Length > 0)),
+            ErrorUtilities.VerifyThrow(((assemblyName?.Length > 0)) || ((assemblyFile?.Length > 0)),
```
Comment (Forgind) [inline src/Shared/AssemblyLoadInfo.cs:26]: extra parens

File: src/Shared/ResourceUtilities.cs [hunk 35]
```diff
@@ -227,7 +227,7 @@ internal static string FormatString(string unformatted, params object[] args)
             string formatted = unformatted;
 
             // NOTE: String.Format() does not allow a null arguments array
-            if ((args != null) && (args.Length > 0))
+            if ((args?.Length > 0))
```
Comment (Forgind) [inline src/Shared/ResourceUtilities.cs:230]: extra parens

File: src/Shared/TaskLoader.cs [hunk 36]
```diff
@@ -181,7 +181,7 @@ internal static bool IsTaskClass(Type type, object unused)
         /// </summary>
         internal static Assembly AssemblyResolver(object sender, ResolveEventArgs args)
         {
-            if ((s_resolverLoadedType != null) && (s_resolverLoadedType.LoadedAssembly != null))
+            if ((s_resolverLoadedType?.LoadedAssembly != null))
```
Comment (Forgind) [inline src/Shared/TaskLoader.cs:184]: extra parens

File: src/Shared/XmlUtilities.cs [hunk 37]
```diff
@@ -49,11 +49,11 @@ internal static XmlElementWithLocation RenameXmlElement(XmlElementWithLocation o
                 newElement.AppendChild(oldElement.FirstChild);
             }
 
-            if (oldElement.ParentNode != null)
-            {
+               
+
```
Comment (Forgind) [inline src/Shared/XmlUtilities.cs:53]: extra spaces

File: src/Tasks/Culture.cs [hunk 38]
```diff
@@ -57,7 +57,7 @@ string dependentUponFilename
 
                 // See if this is a valid culture name.
                 bool validCulture = false;
-                if ((cultureName != null) && (cultureName.Length > 1))
+                if ((cultureName?.Length > 1))
```
Comment (Forgind) [inline src/Tasks/Culture.cs:60]: extra parens

File: src/Tasks/GenerateResource.cs [hunk 39]
```diff
@@ -2431,7 +2431,7 @@ internal bool StronglyTypedResourceSuccessfullyCreated
 #if !FEATURE_ASSEMBLYLOADCONTEXT
             // If references were passed in, we will have to give the ResxResourceReader an object
             // by which it can resolve types that are referenced from within the .RESX.
-            if ((_assemblyFiles != null) && (_assemblyFiles.Length > 0))
+            if ((_assemblyFiles?.Length > 0))
```
Comment (Forgind) [inline src/Tasks/GenerateResource.cs:2434]: extra parens

File: src/Tasks/RegisterAssembly.cs [hunk 40]
```diff
@@ -72,7 +72,7 @@ public override bool Execute()
 
             AssemblyRegistrationCache cacheFile = null;
 
-            if ((AssemblyListFile != null) && (AssemblyListFile.ItemSpec.Length > 0))
+            if ((AssemblyListFile?.ItemSpec.Length > 0))
```
Comment (Forgind) [inline src/Tasks/RegisterAssembly.cs:75]: extra parens

File: src/Tasks/RegisterAssembly.cs [hunk 41]
```diff
@@ -89,7 +89,7 @@ public override bool Execute()
                         string tlbPath;
 
                         // if the type lib path is not supplied, generate default one
-                        if ((TypeLibFiles[i] != null) && (TypeLibFiles[i].ItemSpec.Length > 0))
+                        if ((TypeLibFiles[i]?.ItemSpec.Length > 0))
```
Comment (Forgind) [inline src/Tasks/RegisterAssembly.cs:92]: extra parens

File: src/Tasks/ResolveComReference.cs [hunk 42]
```diff
@@ -327,7 +327,7 @@ public override bool Execute()
 
             _timestampCache = (ResolveComReferenceCache)StateFileBase.DeserializeCache(StateFile, Log, typeof(ResolveComReferenceCache));
 
-            if (_timestampCache == null || (_timestampCache != null && !_timestampCache.ToolPathsMatchCachePaths(_tlbimpPath, _aximpPath)))
+            if (_timestampCache == null || (_timestampCache?.ToolPathsMatchCachePaths(_tlbimpPath, _aximpPath) == false))
```
Comment (Forgind) [inline src/Tasks/ResolveComReference.cs:330]:
```suggestion
            if (_timestampCache?.ToolPathsMatchCachePaths(_tlbimpPath, _aximpPath) != true))
```

File: src/Tasks/UnregisterAssembly.cs [hunk 43]
```diff
@@ -78,7 +78,7 @@ public override bool Execute()
                 for (int i = 0; i < Assemblies.Length; i++)
                 {
                     // if the type lib path is not supplied, generate default one
-                    if (TypeLibFiles != null && TypeLibFiles[i] != null && TypeLibFiles[i].ItemSpec.Length > 0)
+                    if (TypeLibFiles != null && TypeLibFiles[i]?.ItemSpec.Length > 0)
```
Comment (Forgind) [inline src/Tasks/UnregisterAssembly.cs:81]:
```suggestion
                    if (TypeLibFiles?[i]?.ItemSpec.Length > 0)
```

File: src/Xunit.NetCore.Extensions/Discoverers/ConditionalTestDiscoverer.cs [hunk 44]
```diff
@@ -152,11 +152,11 @@ internal static MethodInfo LookupConditionalMethod(Type t, string name)
             TypeInfo ti = t.GetTypeInfo();
 
             MethodInfo mi = ti.GetDeclaredMethod(name);
-            if (mi != null && mi.IsStatic && mi.GetParameters().Length == 0 && mi.ReturnType == typeof(bool))
+            if (mi?.IsStatic == true && mi.GetParameters().Length == 0 && mi.ReturnType == typeof(bool))
                 return mi;
 
             PropertyInfo pi = ti.GetDeclaredProperty(name);
-            if (pi != null && pi.PropertyType == typeof(bool) && pi.GetMethod != null && pi.GetMethod.IsStatic && pi.GetMethod.GetParameters().Length == 0)
+            if (pi != null && pi.PropertyType == typeof(bool) && pi.GetMethod?.IsStatic == true && pi.GetMethod.GetParameters().Length == 0)
```
Comment (Forgind) [inline src/Xunit.NetCore.Extensions/Discoverers/ConditionalTestDiscoverer.cs:159]:
```suggestion
            if (pi?.PropertyType == typeof(bool) && pi.GetMethod?.IsStatic == true && pi.GetMethod.GetParameters().Length == 0)
```

File: src/Deprecated/Conversion/ProjectFileConverter.cs [hunk 45]
```diff
@@ -372,9 +372,9 @@ private void DoConvert()
         {
             // Make sure we were passed in non-empty source and destination project
             // file names.
-            error.VerifyThrowArgument((this.oldProjectFile != null) && (this.oldProjectFile.Length > 0),
+            error.VerifyThrowArgument((this.oldProjectFile?.Length > 0),
                 "MissingOldProjectFile");
-            error.VerifyThrowArgument((this.newProjectFile != null) && (this.newProjectFile.Length > 0),
+            error.VerifyThrowArgument((this.newProjectFile?.Length > 0),
```
Comment (Forgind) [inline src/Deprecated/Conversion/ProjectFileConverter.cs:377]: extra parens * the rest of this file

**Comment by elachlan** [comment]
> I generally like this sort of change because it makes the code cleaner and easier to read—thanks for that!
> 
> I tried to point out all the times this led to unnecessary parentheses or spaces, which constitutes almost all the comments I left. My only worry with this sort of PR is that there could have been one time when you made a mistake (I'm guessing you automated some/all of this?), and I might not have noticed, but I did try to look through fairly carefully and didn't see anything.
> 
> Tentatively approve

Yes, I used Roslynator. I'd like to address the extra parenthesis in another PR. In that PR I'd also look at add parenthesis where needed as well. It would be across the whole solution.

**Comment by elachlan** [comment]
@Forgind this needs test re-run.

**Comment by BenVillalobos** [comment]
Closing to reopen and trigger a CI build. See https://github.com/dotnet/msbuild/issues/5646

**Pull request merged**